app/controllers/api/maintaining/concept_exercises_controller.rb

```diff
@@ -0,0 +1,67 @@
+module API
+  module Maintaining
+    class ConceptExercisesController < BaseController
+      def create
+        example_filename_regexp = "[^a-zA-Z0-9._-]"
+        exercise_slug_regexp = "[^a-zA-Z0-9-]"
+
+        unless params[:exercise_slug].present?
+          return render_error(
+            :missing_exercise_slug,
+            "Exercise Slug is missing"
```

Comment:
For consistency with the later error message:

```suggestion
            "Exercise slug is missing"
```

Reply:
Would it make sense to list the exact API field that is missing? Something like:

```suggestion
            "Exercise Slug is missing. Please post it in the 'missing_exercise_slug' field."
```

Or something similar.

Reply:
Maybe yeah. It depends on whether you want to put the errors here or in the client. I'd potentially delete the message altogether here, and just maintain the text in the client (based on the "type" the API returns) as it probably needs to correspond to the UI, not the API.